dateparser/languages/validation.py

```diff
@@ -0,0 +1,310 @@
+# -*- coding: utf-8 -*-
+import re
+from logging import getLogger
+
+
+class LanguageValidator(object):
```

Comment:
I agree that validation is large and complicated but let me disagree with some other points.

1. Sometimes duplication is necessary. I once work on a project where we had same permission set for posts and comments. So one guy from our team once decided "Hey, it is duplication, lets use one function for both", he refactor the code and half a year later when all code points to new function we had to add an additional permission just for posts and not for comments. Guess what happened :grinning: . Same thing is here: while most of the validation is for lists, it all could change at some point. We can require different validation for weekdays for example.

2. It is actually not re-parsing the file, but validate already extracted data. File is loaded and transformed to yaml by `LanguageDataLoader`

3. Initial purpose of this change was to provide easy readable way to define languages. XML is not that. While it is great for machine-communication, it is rather hard to read for a human. Also i am not sure if it is possible to define facet that would check that groups in replacement are properly correlated to the matching regex.

4. Informative message is just a bonus here, main purpose is to keep all language information consistent. I mean if changes in structure of languages are made it is better to make a proper changes in validator to be sure that all languages received this change and nothing was left out. But validator is actually the one part of code that changes least frequently.

5. Assertions in the code instead of aside validation is really bad idea: parsing code is complicated enough by itself, and validation code does not needed for work. It is best to have validation in separate place.

Someday we can refactor this class to tests suite, but it should be done much-much later on the roadmap. For now we should be considered very young project that does not know what waits around the corner. And to be prepared for the rapid changes we sometimes need messy code. At least until we support the majority of languages around the world and know more about how they treat dates and do not make unnecessary assumptions (like word tokens does not contain spaces or only consists of word-characters) and then there would be less surprises on the way and we can relax a bit :smiley: .

I hope i have managed to explain now why this approach was chosen.